File: 2025-06-10/Two_Sum_II___Input_Array_Is_Sorted.java
package patterns.two_pointers;

import java.util.Arrays;

public class Two_Sum_II___Input_Array_Is_Sorted {
    public static int[] twoSum(int[] numbers, int target) {
        int left = 0, right = numbers.length-1;

        while(left < right) {
            int sum = numbers[left] + numbers[right];
            if(sum == target) {
                return new int[]{left+1, right+1};
            }

            if(sum > target) {
                right--;
            }
            else {
                left++;
            }
        }

        return new int[0];
    }
    public static void main(String[] args) {
        int[] arr = {2,7,11,15};
        System.out.println(Arrays.toString(twoSum(arr, 9)));
    }
}

File: 2026-01-25/Two_Sum_II_Input_Array_Is_Sorted.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if not numbers:
            return []

        l, r = 0, len(numbers)-1
        if target < numbers[0] * 2 or target > numbers[r] * 2:
            return []

        while l < r:
            sum = numbers[l] + numbers[r]
            if sum == target:
                return [l+1, r+1]

            if sum > target:
                r -= 1
            else:
                l += 1

        return [1, 1]


sol = Solution()
print(sol.twoSum([2,7,11,15], 9))